users/tests/tests.py

```diff
@@ -1,42 +1,77 @@
-        self.assertEqual(response.status_code, 302)
+        response = self.client.post(reverse('users:register'), {'username': 'new_user', 'password1': 'pass1234', 'password2': 'pass1234'})
+        self.assertRedirects(response, reverse('users:login'))
 
     def test_register_successful(self):
-        response = self.client.post(reverse('users:register'), {'username': 'user_1', 'password1': 'pass1234', 'password2': 'pass1234'}, follow=True)
-        self.assertIn('You successfully created a new account!', response.content)
+        response = self.client.post(reverse('users:register'), {'username': 'new_user', 'password1': 'pass1234', 'password2': 'pass1234'})
+        self.assertRedirects(response, reverse('users:login'))
 
     def test_register_unsuccessful(self):
-        UserFactory(username='existing_user')
-        response = self.client.post(reverse('users:register'), {'username': 'existing_user', 'password1': 'pass1234', 'password2': 'pass1234'})
-        self.assertIn('A user with that username already exists.', response.content)
+        user = self.create_user('existing_user')
+        response = self.client.post(reverse('users:register'), {'username': user.username, 'password1': 'pass1234', 'password2': 'pass1234'})
+        self.assertTemplateUsed(response, 'registration.html')
+
+
+class UserLoginIntegrationTests(CreateUserMixin, TestCase):
+    def test_redirection_on_login_successful(self):
+        user = self.create_user('existing_user')
+        response = self.client.post(reverse('users:login'), {'username': user.username, 'password': self.password})
+        self.assertRedirects(response, reverse('users:user-profile'))
+
+    def test_login_unsuccessful(self):
+        response = self.client.post(reverse('users:login'), {'username': 'non_existing_user', 'password': 'pass1234'})
+        self.assertTemplateUsed(response, 'login.html')
+
+
+class UserProfileIntegrationTests(CreateUserMixin, TestCase):
+    def test_user_profile_page_with_authenticated_user(self):
+        user = self.create_user('existing_user')
+        self.client.post(reverse('users:login'), {'username': user.username, 'password': self.password})
+        response = self.client.get(reverse('users:user-profile'))
+        self.assertTemplateUsed(response, 'user-profile.html')
+
+
+class AnonymousUserIntegrationTests(TestCase):
+    def test_redirection_on_anonymous_user_access_profile_page(self):
+        response = self.client.get(reverse('users:user-profile'))
+        self.assertEqual(response.status_code, 302)
+
+    def test_redirection_to_login_on_anonymous_user_access_profile_page(self):
+        response = self.client.get(reverse('users:user-profile'))
+        self.assertRedirects(
+            response,
+            '{redirect_path}?next={next_path}'.format(
+                redirect_path=reverse('users:login'),
+                next_path=reverse('users:user-profile')
+            )
+        )
```

Comment:
Join the two test methods. They're testing the same thing. If you're having `assertRedirects`, you shouldn't need to test the status code.